### PR TITLE
fix(dinero): Round intermediate divide and multiply calculations

### LINKED ISSUE #11: Multiplication and division that produce non-integer values throw an Error
Simple example:

```js
const example = { amount: 500, currency: 'USD' }

Dinero(example).divide(3) // Throws
Dinero(example).divide(2) // Returns valid Dinero object

Dinero(example).multiply(2.01) // Throws
Dinero(example).multiply(2) // Returns valid Dinero object
```

I assume there should be some intermediate rounding that should be happening, and I'd be happy to write a PR that adds some property-based tests and a rounding solution, but wanted a little bit of direction on precision here.

### PR DESCRIPTION
**Pull request by scotttrinh**
Use [Banker's Rounding](http://wiki.c2.com/?BankersRounding) to round intermediate calculations in `#divide` and `#multiply`. Added a few property-based tests to add some robustness to the expectations and test some edge cases.

fix #11

### CODE REVIEW

File: src/services/helpers.js [hunk 1]
```diff
@@ -9,3 +9,12 @@
 export function isNumeric(value) {
   return !isNaN(parseInt(value)) && isFinite(value)
 }
+
+export function bankersRound(x) {
```
[the remark this answers is not in the document]
Comment (scotttrinh) [reply src/services/helpers.js]: What would you expect `precision` to do? We're rounding to integers here, and Banker's Round only actually does something different for values that lie directly between two integers, right?

File: src/services/helpers.js [hunk 2]
```diff
@@ -9,3 +9,12 @@
...
+  var r = Math.round(x)
+  return (x > 0 ? x : -x) % 1 === 0.5 ? (even(r) ? r : r - 1) : r
+}
+
+function even(r) {
```
[the remark this answers is not in the document]
Comment (scotttrinh) [reply src/services/helpers.js]: Added some property-based tests to this, too. I'm always suspect of `%` since it's really remainder not modulo and I always worry about the ordering. 😅

[the remark this answers is not in the document]
Comment (sarahdayan) [reply src/services/helpers.js]: @scotttrinh Did you amend your commit? I can see your changes but GitHub doesn't detect a new commit.

[the remark this answers is not in the document]
Comment (scotttrinh) [reply src/services/helpers.js]: That's correct. I wasn't quite what to say in the `cz` flow, so I figured these changes don't really affect the actual content of the commit. If you'd prefer a more interesting commit history, I can break down this PR into a series of more granular commits.

[the remark this answers is not in the document]
Comment (sarahdayan) [reply src/services/helpers.js]: No it's cool 👍

File: src/services/calculator.js [hunk 3]
```diff
@@ -59,6 +61,22 @@ export default function Calculator() {
...
     modulo(a, b) {
       return a % b
+    },
+
+    /**
+     * Returns a rounded number where if the input number is equidistant from the
+     * two nearest integers it is rounded to the nearest even integer.
+     * @ignore
+     *
+     * @param  {Number} value - The number to round
```
Comment (sarahdayan) [inline src/services/calculator.js]: Full stop is missing at the end.

File: src/services/calculator.js [hunk 4]
```diff
@@ -59,6 +61,22 @@ export default function Calculator() {
...
+     *
+     * @return {Number}
+     */
+    bankersRound(value) {
+      var rounded = Math.round(value)
```
Comment (sarahdayan) [inline src/services/calculator.js]: Use `const` over `var`. I'll enforce this in ESLint config later as well.

File: src/services/calculator.js [hunk 5]
```diff
@@ -59,6 +61,22 @@ export default function Calculator() {
      */
     modulo(a, b) {
       return a % b
+    },
+
```
Comment (sarahdayan) [inline src/services/calculator.js]: No need for the blank line.

File: src/dinero.js [hunk 6]
```diff
@@ -219,7 +221,9 @@ const Dinero = options => {
      */
     divide(divisor) {
       return create.call(this, {
-        amount: calculator.divide(this.getAmount(), divisor)
+        amount: calculator.bankersRound(
```
Comment (sarahdayan) [inline src/dinero.js]: Would be great to add a note and example in the doc block.

```
/**
  * Returns a new Dinero object that represents the divided value by the given factor.
  *
  * [Banker's rounding](http://wiki.c2.com/?BankersRounding) is used to handle fractional cents.
  * This *can* lead to accuracy issues as you chain many times. Consider a minimal amount of subsequent calculations for safer results.
  *
  * @param  {Number} divisor - The factor to divide by.
  *
  * @example
  * // returns a Dinero object with amount 100
  * Dinero({ amount: 400 }).divide(4)
  * // returns a Dinero object with amount 52
  * Dinero({ amount: 105 }).divide(2)
  *
  * @return {Dinero}
  */
```

File: src/dinero.js [hunk 7]
```diff
@@ -203,7 +203,9 @@ const Dinero = options => {
      */
     multiply(multiplier) {
       return create.call(this, {
-        amount: calculator.multiply(this.getAmount(), multiplier)
+        amount: calculator.bankersRound(
```
Comment (sarahdayan) [inline src/dinero.js]: Would be great to add a note and example in the doc block.

```
/**
  * Returns a new Dinero object that represents the multiplied value by the given factor.
  *
  * [Banker's rounding](http://wiki.c2.com/?BankersRounding) is used to handle fractional cents.
  * This *can* lead to accuracy issues as you chain many times. Consider a minimal amount of subsequent calculations for safer results.
  *
  * @param  {Number} multiplier - The factor to multiply by.
  *
  * @example
  * // returns a Dinero object with amount 1600
  * Dinero({ amount: 400 }).multiply(4)
  * // returns a Dinero object with amount 800
  * Dinero({ amount: 400 }).multiply(2.001)
  *
  * @return {Dinero}
  */
```

**Comment by scotttrinh** [comment]
@sarahdayan Thanks! The big internet provider in NYC just went down, so I'll get these changes incorporated ASAP!
